models/issue.go

```diff
@@ -1901,6 +1899,7 @@ func (issue *Issue) getBlockedByDependencies(e Engine) (issueDeps []*DependencyI
 		Join("INNER", "repository", "repository.id = issue.repo_id").
 		Join("INNER", "issue_dependency", "issue_dependency.dependency_id = issue.id").
 		Where("issue_id = ?", issue.ID).
+		OrderBy("repository.id, issue.id ASC").
```

Comment:
Mmm.... If you sort by issue.id DESC you get the newer issues first, which would make sense. Order by repository ID makes no sense. It's an internal number.
Perhaps:
```
OrderBy("CASE WHEN repository.id = ? THEN 1 ELSE 2 END, issue.id DESC", current-repo-id)
```
would cause the issues from the current repo to show first.

Reply:
gotcha, these are for the list of blocked by or blocking dependencies displayed on the issue page (i.e. not the drop-down list to add a new one). I was thinking it would make sense for the issues to be in order of their id grouped by repo, so you'd see all the "repo1" issues and then "repo2" issues, not mixed together. But I agree with your suggestion, thanks!

Reply:
Oh, I see! To group by repo and still have the current repo top of the list:
```
OrderBy("CASE WHEN repository.id = ? THEN 0 ELSE repository.id END, issue.id DESC", current-repo-id)
```
But I've got confused. I thought this was for the search function.

Reply:
thanks for the suggestion, I used the order by statement you suggested